packages/admin-ui/src/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+import { makeDecoratable, withStaticProps } from "~/utils";
+import { DialogContent } from "./components/DialogContent";
+import { DialogHeader } from "~/Dialog/components/DialogHeader";
+import { DialogBody } from "~/Dialog/components/DialogBody";
+import { DialogFooter } from "~/Dialog/components/DialogFooter";
+import { DialogOverlay } from "~/Dialog/components/DialogOverlay";
+import { DialogPortal } from "./components/DialogPortal";
+import { DialogRoot } from "./components/DialogRoot";
+import { DialogTrigger } from "./components/DialogTrigger";
+import { TitleIcon } from "./components/TitleIcon";
+import { ConfirmButton } from "./components/ConfirmButton";
+import { CancelButton } from "./components/CancelButton";
+
+export interface DialogProps
+    extends React.ComponentPropsWithoutRef<typeof DialogRoot>,
+        Omit<React.ComponentPropsWithoutRef<typeof DialogContent>, "title"> {
+    trigger?: React.ReactNode;
+    title?: React.ReactNode | string;
+    titleIcon?: React.ReactElement;
+    showCloseButton?: boolean;
+    preventOutsideDismiss?: boolean;
+    bodyPadding?: boolean;
+    description?: React.ReactNode | string;
```

Comment:
`React.ReactNode` already includes `string`, right? No need to add it.

Reply:
Nice one, ty!